engine/api/src/main/java/io/deephaven/engine/table/TableUpdate.java

```diff
@@ -52,25 +72,22 @@ && modified().isEmpty()
     }
 
     /**
-     * @return true if all internal state is initialized
+     * Throws an {@link io.deephaven.base.verify.AssertionFailure} if this update is not structurally valid, i.e. its
+     * accessors will return non-{@code null}, usable data structures. This does not test for usage outside the
+     * appropriate updating phase.
      */
-    default boolean valid() {
-        return added() != null
-                && removed() != null
-                && modified() != null
-                && shifted() != null
-                && modifiedColumnSet() != null;
-    }
+    void validate();
```

Comment:
I have made two comments. I do wonder if we want `AssertionFailure` to be part of the API for this method. Alternatively, we could introduce a new exception class.

Reply:
We want an assertion failure, it can be used to trigger a heap dump.